ragna/deploy/_api/core.py

```diff
@@ -210,6 +210,30 @@ async def upload_document(
 
             return document
 
+    @app.get("/documents")
+    async def get_documents(user: UserDependency) -> list[schemas.Document]:
+        with get_session() as session:
+            return database.get_documents(session, user=user)
+
+    @app.get("/documents/{id}")
+    async def get_document(user: UserDependency, id: uuid.UUID) -> schemas.Document:
+        with get_session() as session:
+            document, _ = database.get_document(session, user=user, id=id)
+            return document
+
+    @app.get("/documents/{id}/content")
+    async def get_document_content(user: UserDependency, id: uuid.UUID) -> bytes:
+        with get_session() as session:
+            _, metadata = database.get_document(session, user=user, id=id)
+            if "path" not in metadata:
+                raise HTTPException(
+                    status_code=400,
+                    detail="Document path not found",
+                )
+            with aiofiles.open(metadata["path"], "rb") as file:
+                while content := await file.read(1024):
+                    yield content
```

Comment:
Our documents have a `.read()` method that should be used here:

https://github.com/Quansight/ragna/blob/2066dcdeb949060873a8552449b848b245e0e985/ragna/core/_document.py#L75-L77

It is not async for now, but there is no need to require documents to be on the local file system. We just need the conversion to the `Document` object

https://github.com/Quansight/ragna/blob/2066dcdeb949060873a8552449b848b245e0e985/ragna/deploy/_api/schemas.py#L29-L37

Maybe this is also a good time to resolve this TODO first?